### PR TITLE
Better caching

### PR DESCRIPTION
**Pull request by elegaanz**
Based on @trinity-1686a's patch.

I'm not sure it is a fix for #786 but it is better than nothing.

### CODE REVIEW

**Comment by elegaanz** [comment]
OK, my git history is broken, I need to fix that.

File: src/template_utils.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ impl<'r> Responder<'r> for Ructe {
         let etag = format!("{:x}", hasher.finish());
         if r.headers()
             .get("If-None-Match")
-            .any(|s| s[1..s.len() - 1] == etag)
+            .any(|s| s[1..s.len() - 1] == etag || s[3..s.len() - 1] == etag)
```
Comment (igalic) [inline src/template_utils.rs:66]: can one of you two explain what this does?

**Comment by igalic** [comment]
don't we still have to to change the subscribe/follow/blah button from a form… to… well… a button?

File: src/template_utils.rs (hunk 1 above)
Comment (trinity-1686a) [reply src/template_utils.rs:66]: this should probably get it's comment actually.
We send Etags as `'some_tag'`, the old match does not verify the `'`, but check that whatever in-between is the right tag. Nginx apparently change the Etag to `W/'some_tag'`, W/ indicating it's a weak Etag, the content may not be byte-equal, but is semantically identical (which is imo a shitty but correct interpretation of the RFC, as Nginx send compressed data)

The new match ignore that W/ saying it's a weak etag and so match if Nginx modified the header we sent previously

Comment might looks like : // if a strong or weak etag matches

**Comment by trinity-1686a** [comment]
well there is that too, but these buttons are not here on the front page or a blog or user page. Anyway for post a better caching solution that does not require to query the database should be added

**Review by FreyjaWildes — APPROVED** [review]
I'm not totally sure, being new to the software and Rust, but this seems fair enough and a small change. Probably can be approved, right ?

**Comment by KitaitiMakoto** [comment]
@FreyjaWildes You're right. I merged this pull request as https://git.joinplu.me/Plume/Plume/pulls/840

Thank you for keeping interest on Plume!

**Pull request merged**